compiler/rustc_codegen_llvm/src/llvm/mod.rs

```diff
@@ -41,6 +41,32 @@ pub(crate) fn AddFunctionAttributes<'ll>(
     }
 }
 
+pub(crate) fn HasAttributeAtIndex<'ll>(
+    llfn: &'ll Value,
+    idx: AttributePlace,
+    kind: AttributeKind,
+) -> bool {
+    unsafe { LLVMRustHasAttributeAtIndex(llfn, idx.as_uint(), kind) }
+}
+
+pub(crate) fn HasStringAttribute<'ll>(llfn: &'ll Value, name: *const i8) -> bool {
+    unsafe { LLVMRustHasFnAttribute(llfn, name) }
+}
+
+pub(crate) fn RemoveStringAttrFromFn<'ll>(llfn: &'ll Value, name: *const i8) {
+    unsafe { LLVMRustRemoveFnAttribute(llfn, name) }
```

Comment:
@oli-obk Not for this PR, but shouldn't we at some point add some debug assertion to verify all those functions, instead of just silently ignoring that these wrappers make it trivial to achieve UB from safe code?
E.g. check that name is zero terminated, Value is actually a function, etc.
(That's not only about autodiff, every function here does that.)